x-pack/plugins/security_solution/public/common/components/filter_group/index.tsx

```diff
@@ -333,10 +324,53 @@ const FilterGroupComponent = (props: PropsWithChildren<FilterGroupProps>) => {
     setShowFiltersChangedBanner(false);
   }, [controlGroup, switchToViewMode, getStoredControlInput, hasPendingChanges]);
 
+  const upsertPersistableControls = useCallback(() => {
+    const persistableControls = initialControls.filter((control) => control.persist === true);
+    if (persistableControls.length > 0) {
+      const currentPanels = controlGroup?.getInput().panels;
+      const orderedPanels = (
+        Object.values(currentPanels ?? []) as Array<ControlPanelState<OptionsListEmbeddableInput>>
+      ).sort((a, b) => a.order - b.order);
+      persistableControls.forEach((control) => {
+        const controlExists = (
+          Object.values(currentPanels ?? []) as Array<ControlPanelState<OptionsListEmbeddableInput>>
+        ).findIndex((currControl) => control.fieldName === currControl.explicitInput.fieldName);
+        if (controlExists === -1) {
```

Comment:
Same thing here, you can use `orderedPanels.some`

Reply:
Alternatively, I'm not a huge fan of side effects, but there's some nested looping that might be avoidable, if you keep a temp map from the sort on line 333:

But totally fine if you don't do this as I don't expect the array to get large enough where this ever becomes a significant performance hit
```
currControlMap = { }

const orderedPanels = (
Object.values(currentPanels ?? []) as Array<ControlPanelState<OptionsListEmbeddableInput>>)
.sort((a, b) => {
  if (a.explicitInput.fieldName) currControlMap[a.explicitInput.fieldName] = true;
  return a.order - b.order;
});

```

then here,

```
persistableControls.forEach((control) => {
   const controlExists =  currControlMap[control.fieldName];
   if (controlExists) { .... }
}
```

Reply:
Thanks for checking in details. Let me refactor this. I agree with the use of `some` and avoiding to call `Object.values` twice.

Additionally, nested loop was not event needed, refactored that as well.